darshan-util/pydarshan/tests/test_plot_dxt_heatmap.py

```diff
@@ -0,0 +1,458 @@
+    hmap_positions = jgrid.ax_joint.get_position()
+    assert hmap_positions.x0 == 0.1
+    assert hmap_positions.y0 == 0.15000000000000002
+    assert hmap_positions.y1 == 0.774390243902439
+    if filepath == "examples/example-logs/ior_hdf5_example.darshan":
+        # since `ior_hdf5_example.darshan` has 4 ranks, it has
+        # different x max values because it needs room for
+        # the colorbar on the outside of the horizontal bar graph
+        assert hmap_positions.x1 == 0.7158709677419354
+    else:
+        assert hmap_positions.x1 == 0.7824516129032258
+
+    # get vertical bar graph positions
+    vert_bar_positions = jgrid.ax_marg_x.get_position()
+    assert vert_bar_positions.x0 == 0.1
+    assert vert_bar_positions.y0 == 0.7780487804878049
+    assert vert_bar_positions.y1 == 0.9
+    if filepath == "examples/example-logs/ior_hdf5_example.darshan":
+        # since `ior_hdf5_example.darshan` has 4 ranks, the vertical
+        # bar graph has a different x max value because it needs room for
+        # the colorbar on the outside of the horizontal bar graph
+        assert vert_bar_positions.x1 == 0.7158709677419354
+    else:
+        assert vert_bar_positions.x1 == 0.7824516129032258
+
+    # get horizontal bar graph positions
+    horiz_bar_positions = jgrid.ax_marg_y.get_position()
+    assert horiz_bar_positions.y0 == 0.15000000000000002
+    assert horiz_bar_positions.y1 == 0.774390243902439
+    if filepath == "examples/example-logs/ior_hdf5_example.darshan":
+        # since `ior_hdf5_example.darshan` has 4 ranks, the horizontal
+        # bar graph has different x min/max values because it has to
+        # make room for the colorbar
+        assert horiz_bar_positions.x0 == 0.7206451612903225
+        assert horiz_bar_positions.x1 == 0.84
+    else:
+        assert horiz_bar_positions.x0 == 0.7877419354838711
+        assert horiz_bar_positions.x1 == 0.92
+
+    # get the colorbar positions
+    cbar_positions = jgrid.fig.axes[-1].get_position()
+    assert cbar_positions.y0 == 0.15000000000000002
+    assert cbar_positions.y1 == 0.774390243902439
+    if filepath == "examples/example-logs/ior_hdf5_example.darshan":
+        # since `ior_hdf5_example.darshan` has 4 ranks, the colorbar has
+        # to go closer to the edge of the figure so the horizontal bar
+        # graph can fit in the panel
+        assert cbar_positions.x0 == 0.85
+        assert cbar_positions.x1 == 0.8716135084427767
+    else:
+        assert cbar_positions.x0 == 0.82
+        assert cbar_positions.x1 == 0.8416135084427767
+
+
+@pytest.mark.parametrize(
+    "filepath",
+    [
+        "examples/example-logs/ior_hdf5_example.darshan",
+        "examples/example-logs/dxt.darshan",
+        "tests/input/sample-dxt-simple.darshan",
+    ],
+)
+@pytest.mark.parametrize("mods", [["DXT_POSIX"], ["DXT_MPIIO"]])
+@pytest.mark.parametrize("ops", [["read", "write"], ["read"], ["write"]])
+def test_plot_heatmap(filepath, mods, ops):
+
```

Comment:
I had a difficult time coming up with meaningful checks here (for `plot_dxt_heatmap.plot_heatmap()`). I feel like most of these checks are frivolous, outside of testing the proper errors are raised.